### PR TITLE
Explainer: Add a more general introduction / motivation section.

### PR DESCRIPTION
**Pull request by Kangz**
PTAL!

### CODE REVIEW

**Review by kainino0x — APPROVED** [review]

File: explainer/index.bs [hunk 1]
```diff
@@ -23,19 +23,59 @@ Issue(tabatkins/bikeshed#2006): Set up cross-linking into the WebGPU and WGSL sp
...
-[more actual motivation](https://w3ctag.github.io/explainers)
-is needed (here, in the spec Introduction section, or both).
+## Use cases ## {#use-cases}
 
+Example use cases for WebGPU that aren't addressed by WebGL 2 are:
+
+ - Drawing images with highly-detailed scenes with many different objects (such as CAD models). WebGPU's drawing commands are individually cheaper than WebGL's.
+ - Execute advanced algorithms for drawing realistic scenes. Many modern rendering techniques and optimizations cannot execute on WebGL 2 due to the lack of support for general computations.
+ - Execute machine learning models efficiently on the GPU. WebGL can be massaged to execute machine learning but it is difficult and sub-optimal.
+ - Provide a foundation to expose modern GPU functionality on the Web. WebGPU exposes functionality that became available in native around 2010 but is structured similarly to all current native GPU APIs. There are plans to later extend it to have more modern functionality. See also: [[#why-not-webgl3]].
```
Comment (kvark) [inline explainer/index.bs:39]: this reads wrong. Was it meant to talk about WebGL and not WebGPU?

File: explainer/index.bs [hunk 2]
```diff
@@ -23,19 +23,59 @@ Issue(tabatkins/bikeshed#2006): Set up cross-linking into the WebGPU and WGSL sp
...
+
+Concrete examples are:
+
+ - Improving existing Javascript 3D libraries like Babylon.js and Three.js with new rendering techniques (compute-based particles, fancier post-processing, ...) and offloading on the GPU expensive computations currently done one the CPU (culling, skinned model transformation, ...)
```
Comment (kvark) [inline explainer/index.bs:43]: nit: missing period at the end

File: explainer/index.bs [hunk 3]
```diff
@@ -23,19 +23,59 @@ Issue(tabatkins/bikeshed#2006): Set up cross-linking into the WebGPU and WGSL sp
...
+ - Porting newer game engines to the Web, and enable engines to expose more advanced rendering features. For example Unity's WebGL export uses the "mobile" feature set, but WebGPU could use a higher feature set.
```
Comment (kvark) [inline explainer/index.bs:44]: Not sure I understand the bit about Unity's WebGL. If it can't target some mobile GPUs because of their limits of features, how is WebGPU going to help this case?

File: explainer/index.bs [hunk 4]
```diff
@@ -23,19 +23,59 @@ Issue(tabatkins/bikeshed#2006): Set up cross-linking into the WebGPU and WGSL sp
...
+ - Enable general purpose computations to be executed efficiently on the GPU.
+ - Support implementation targeting various native GPU APIs: Microsoft's D3D12, Apple's Metal and Khronos' Vulkan.
+ - Provide a human-authorable language to specify computations to run on the GPU.
+ - Be implementable in the multi-process architecture of browsers and uphold the security of the Web.
+ - As much as possible, have applications work portably across different user systems and browsers.
+ - Interact with the rest of the Web platform in useful but carefully-scoped ways (essentially sharing images one way or another).
+
+Non-goals:
+
+ - Expose support for non-programmable hardware like DSPs or more fixed-function machine learning hardware.
```
Comment (kvark) [inline explainer/index.bs:62]: GPUs also have non-programmable hardware. Could we say here that we are talking about non-GPU things?

File: explainer/index.bs [hunk 5]
```diff
@@ -23,19 +23,59 @@ Issue(tabatkins/bikeshed#2006): Set up cross-linking into the WebGPU and WGSL sp
...
+ - Expose support for hardware without support for general-purpose computations (like older mobile phones or even older desktop GPUs).
```
Comment (kvark) [inline explainer/index.bs:63]: maybe reword this? "expose support for hardware without support" is a bit repetitive

File: explainer/index.bs [hunk 6]
```diff
@@ -23,19 +23,59 @@ Issue(tabatkins/bikeshed#2006): Set up cross-linking into the WebGPU and WGSL sp
...
+ - Exhaustively expose all functionality available on native GPU APIs.
```
Comment (kvark) [inline explainer/index.bs:64]: may need clarification on why

File: explainer/index.bs (hunk 1 above)
Comment (Kangz) [reply explainer/index.bs:39]: Can you detail what reads wrong? I'm not sure I understand.

File: explainer/index.bs (hunk 3 above)
Comment (Kangz) [reply explainer/index.bs:44]: Rephrased to use "lowest" instead.

File: explainer/index.bs (hunk 4 above)
Comment (Kangz) [reply explainer/index.bs]: Rephrased slightly using "hardware that's not programmable at all".

File: explainer/index.bs (hunk 1 above)
Comment (kvark) [reply explainer/index.bs:39]: It says "WebGPU exposes functionality that became available in native around 2010".
In 2010 there were not any pipelines or passes, so I assume this sentence was meant to talk about WebGL instead.

Comment (Kangz) [reply explainer/index.bs]: This was referring to compute shaders but extremely unclear. Removed that mention and reworded a bit.

**Review by kvark — APPROVED** [review]
